libs/langchain/langchain/document_loaders/sitemap.py

```diff
@@ -70,7 +113,12 @@ def __init__(
 
         super().__init__(web_paths=[web_path], **kwargs)
 
-        self.filter_urls = filter_urls
+        if filter_urls is None and not is_local:
+            self.filter_urls: Optional[List[str]] = [
+                _extract_domain_and_scheme(web_path)
+            ]
```

Comment:
Filter URLs are treated as a regex, so extracting `http://example.com` from the web path will inappropriately match both `http://examplexcom.org` and `http://example.com.attacker.com/`

A better approach would be to parse each target URL and check it against the scheme and domain, then match it against the regex with proper escaping.